### PR TITLE
fix: correct `bn` formatting bug with 0 units

### LINKED ISSUE #3227: `bn` library bugs when you're using units / precision zero
### fuels-ts SDK Version

latest

### Toolchain Versions

not relevant

### Node.js Version

_No response_

### Browser

_No response_

### Operating System

_No response_

### Describe the Problem

when you call `format({ units: 0, precision: 0 })` it will not return the correct value

the same happens on `parseUnits` function

### Code Snippet

```shell
the workaround we've been doing on FE side is:


bn(amount).mul(10).format({ units: 1, precision: 0 });
```

which is basically multiplying by 10 first, then converting to units 1, which at the end of the day will be the same of passing `units: 0` 

this is very overwhelming to do as we have many projects and some places can be forgotten, leading to number bugs.
```


### Contract ABI

_No response_

### Errors

_No response_

### PR DESCRIPTION
**Pull request by maschad**
- Closes #3227

# Release notes

In this release, we:

- fixed an issue where formatting with  `0` in the `bn` class returned an incorrect value

# Summary

Fixed a formatting issue in the `bn` library which caused big numbers formatted with `unit`,  `minPrecision` and/or `precision` set to `0` to be incorrectly formatted.

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

File: packages/math/src/bn.ts [hunk 1]
```diff
@@ -109,37 +108,78 @@ export class BN extends BnJs implements BNInputOverrides, BNHiddenTypes, BNHelpe
   format(options?: FormatConfig): string {
     const {
       units = DEFAULT_DECIMAL_UNITS,
-      precision = DEFAULT_PRECISION,
-      minPrecision = DEFAULT_MIN_PRECISION,
+      precision: initialPrecision = DEFAULT_PRECISION,
+      minPrecision: initialMinPrecision = DEFAULT_MIN_PRECISION,
```
Comment (petertonysmith94) [inline packages/math/src/bn.ts:112]: I believe we should remove `minPrecision` in the future.

Comment (petertonysmith94) [reply packages/math/src/bn.ts:112]: Maybe create an issue around this?

Please resolve at will

**Review by petertonysmith94 — APPROVED** [review]

File: packages/math/src/bn.test.ts [hunk 2]
```diff
@@ -495,4 +495,32 @@ describe('Math - BN', () => {
...
+  it('should format properly with 0 minPrecision', () => {
+    expect(bn('1000000000').format({ minPrecision: 0 })).toEqual('1');
+  });
+
+  it('should properly format with minPrecision 0 and precision 1', () => {
+    expect(bn('10010000').format({ units: 5, minPrecision: 0, precision: 1 })).toEqual('100.1');
+  });
+
+  it('should properly format with minPrecision 0 and precision 1 with a leading zero', () => {
+    expect(bn('100000').format({ units: 5, minPrecision: 0, precision: 1 })).toEqual('1');
```
Comment (nedsalk) [inline packages/math/src/bn.test.ts:520]: The test name says that there should be a leading zero here but the `.toEqual(1)` contains none :question:

Comment (maschad) [reply packages/math/src/bn.test.ts:520]: The test name is incorrect, I should have said `with a trailing zero`

File: packages/math/src/bn.test.ts [hunk 3]
```diff
@@ -495,4 +495,32 @@ describe('Math - BN', () => {
...
+
+  it('should format properly with 0 minPrecision', () => {
+    expect(bn('1000000000').format({ minPrecision: 0 })).toEqual('1');
+  });
+
+  it('should properly format with minPrecision 0 and precision 1', () => {
+    expect(bn('10010000').format({ units: 5, minPrecision: 0, precision: 1 })).toEqual('100.1');
+  });
+
+  it('should properly format with minPrecision 0 and precision 1 with a leading zero', () => {
```
Comment (danielbate) [inline packages/math/src/bn.test.ts:519]:
```suggestion
  it('should properly format with minPrecision 0 and precision 1 with a trailing zero', () => {
```

**Review by danielbate — APPROVED** [review]

**Review by petertonysmith94 — APPROVED** [review]

**Review by Dhaiwat10 — APPROVED** [review]

**Review by arboleya — APPROVED** [review]

**Pull request merged**